src/mongocxx/exception/mongocxx_error.hpp

```diff
@@ -16,21 +16,24 @@
 
 #include <mongocxx/config/prelude.hpp>
 
-#include <bsoncxx/stdx/optional.hpp>
-
-#include <mongocxx/exception/base.hpp>
-#include <mongocxx/stdx.hpp>
+#include <limits>
+#include <system_error>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
-namespace exception {
 
-class MONGOCXX_API operation : public base {
-   public:
-    using base::base;
+enum class mongocxx_error : std::int32_t {
+	k_unknown = std::numeric_limits<std::int32_t>::max(),
```

Comment:
Should I not have added the k_unknown? I'm wondering if the same logic applies to the bsoncxx error handling cr, where we don't have k_unknown since we are creating all the errors, so they should all be known.